web/tests/filter.test.js

```diff
@@ -2446,10 +2454,7 @@ run_test("adjusted_terms_if_moved", () => {
         {operator: "with", operand: `${message.id}`},
     ];
     result = Filter.adjusted_terms_if_moved(raw_terms, message);
-    assert.deepEqual(result, [
-        {operator: "dm", operand: "user3@zulip.com", negated: false},
-        {operator: "with", operand: "2"},
-    ]);
+    assert.deepEqual(result, [{operator: "dm", operand: "user3@zulip.com", negated: false}]);
```

Comment:
How was this working previously ? -- I don't see `with` retained in the url in this case when tested out in CZO.

Reply:
So basically it would check the message id, and since it would update the narrow accordingly, and then remove the `with` term -- same as in https://github.com/zulip/zulip/pull/31239/files#r1721784931

Previously, the `with` was removed before correcting the narrow and now the narrow is being corrected and then `with` is removed for `dm` terms